MotionMark/tests/core/resources/multiply.js

```diff
@@ -24,120 +24,183 @@
+        return this.stage.hslPrefix + l + "%," + influence + ")";
+    }
+
+    animate(timestamp) {
+        this.rotate += this.step;
+        this.roundedRect.style.transform = "rotate(" + this.rotate + "deg)";
+        this.roundedRect.style.backgroundColor = this.backgroundColor();
+    }
+}
+
 var MultiplyStage = Utilities.createSubclass(Stage,
     function()
     {
         Stage.call(this);
         this.tiles = [];
-        this._offsetIndex = 0;
+        this.activeLength = 0;
     }, {
 
-    visibleCSS: [
-        ["display", "none", "block"]
-    ],
-    totalRows: 68,
+    initialRowsCount: 69,
+    extraExpandRows: 16,
 
-    initialize: function(benchmark, options)
-    {
+    initialize: function(benchmark, options) {
         Stage.prototype.initialize.call(this, benchmark, options);
-        var tileSize = Math.round(this.size.height / this.totalRows);
-        if (options.visibleCSS)
-            this.visibleCSS = options.visibleCSS;
-
-        // Fill the scene with elements
-        var x = Math.round((this.size.width - tileSize) / 2);
-        var y = Math.round((this.size.height - tileSize) / 2);
-        var tileStride = tileSize;
-        var direction = 0;
-        var spiralCounter = 2;
-        var nextIndex = 1;
-        var maxSide = Math.floor(y / tileStride) * 2 + 1;
-        this._centerSpiralCount = maxSide * maxSide;
-        for (var i = 0; i < this._centerSpiralCount; ++i) {
-            this._addTile(x, y, tileSize, Stage.randomInt(0, 359));
-
-            if (i == nextIndex) {
-                direction = (direction + 1) % 4;
-                spiralCounter++;
-                nextIndex += spiralCounter >> 1;
-            }
-            if (direction == 0)
-                x += tileStride;
-            else if (direction == 1)
-                y -= tileStride;
-            else if (direction == 2)
-                x -= tileStride;
-            else
-                y += tileStride;
-        }
 
-        this._sidePanelCount = maxSide * Math.floor((this.size.width - x) / tileStride) * 2;
-        for (var i = 0; i < this._sidePanelCount; ++i) {
-            var sideX = x + Math.floor(Math.floor(i / maxSide) / 2) * tileStride;
-            var sideY = y - tileStride * (i % maxSide);
+        this.rowsCount = this.initialRowsCount;
+
+        this.calculateTileSize();
+        this.calculateGridSize();
+        this.iterator = new SpiralIterator(this.tileGrid);
+
+        while (!this.iterator.isDone())
+            this.tiles.push(this.createTile());
+    },
+
+    calculateTileSize: function() {
+        let tileSide = Math.round(this.size.height / this.rowsCount);
+        this.tileSize = new Size(tileSide, tileSide);
+    },
+
+    calculateGridSize: function() {
+        let columnsCount = Math.floor(this.size.width / this.tileSize.width);
+        if (columnsCount % 2 == 0)
+            --columnsCount;
+        this.tileGrid = new Size(columnsCount, this.rowsCount);
```

Comment:
When running this the visual jumps in the size of the grid feel like a bug. Is it possible to keep the visual rectangular size the same?

Reply:
Yes this can be done if the spiral iterator is restarted instead of having it expanded. This will create layers of elements every time the spiral iterator is restarted. The opacity of the rounded rect tile can be a factor of its z-index and its distance from the center. 

But this might be behavioral change and the score can be different. Are we okay with that?

Reply:
I implemented the restarting solution. And I made the rowsCount to be 59 always. Once the test maxes out the grid size it will create a new iterator.

@sky2 please try this solution. If you do not see the layers of elements on your machine change the `rowsCount` to be a smaller value.

Reply:
I like this one better in that the visual size of the grid doesn't jump around. thanks.